### PR TITLE
Handle sysv compat mode when checking enabled status for systemd service

### PR DESCRIPTION
**Pull request by joshuamiller01**
Signed-off-by: Joshua Miller <joshmiller@fb.com>

When deciding on enabled status of a systemd service, if the version of systemd is sufficiently old, and the service is not a native systemd service but instead relies on the systemd-sysv-generator, shellout to `is-enabled` to determine whether the service is enabled.  

## Description
Idempotency for action :enable on systemd service was broken (due to https://github.com/chef/chef/pull/10776), such that every chef run triggered enablement despite the service already being enabled.

## Related Issue
NA

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.
- [X] I have run the pre-merge tests locally and they pass.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tas50** [comment]
@ramereth I believe this fixes your issue

**Comment by ramereth** [comment]
> @ramereth I believe this fixes your issue

Is there a test build artifact I could grab to verify it otherwise I can test by manually patching it on a system?

**Comment by joshuamiller01** [comment]
@ramereth I don't know how / if build artifacts come about.  I did test this by manually patching in this change on a rando ubuntu 16.04 box running chef 16.9.32, systemd 229, and the apache2 non-native service, for whatever that's worth.

**Comment by ramereth** [comment]
@joshuamiller01 Let me do a few local tests using a patch to provide some accurate feedback

**Review by jaymzh — APPROVED** [review]
Thanks!

**Comment by jaymzh** [comment]
will wait for confirmation before merging, but tests pass (failing tests are unrelated) and new tests added, so I'm happy with the PR in general.

**Comment by ramereth** [comment]
@tas50 @joshuamiller01 confirmed this fixes the issue

**Comment by ramereth** [comment]
@tas50 @joshuamiller01 confirmed this fixes the issue

**Pull request merged**
